### PR TITLE
feat: dynamic chart resolution based on selected time period

### PR DESCRIPTION
**Pull request by oleksii-donets**
## Summary
- Automatically adjusts chart time-window resolution (bucket size) based on the selected time range, keeping data points at a sensible density (≤200 points)
- Adds `getChartResolution` utility that picks the smallest "nice" bucket size (1m, 2m, 5m, ... up to 1d) for the given duration
- Applies the computed resolution to telemetry query `window()` calls before fetching dashboard data

## Test plan
- [x] Unit tests added for `getChartResolution` covering all standard time ranges plus edge cases
- [x] Unit tests added for `applyResolutionToQuery` verifying pattern replacement in expressions and groupBy
- [ ] Manual verification: select different time ranges in the dashboard and confirm chart granularity adjusts appropriately

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 # Telemetry
 # GRAFANA_LINK=""
```
Comment (dearsimanovich-epam) [inline .env.template:11]: not 100% sure that we have to make this configurable, I mean in 99,9% it will be default value

Comment (PolinaGurinovich97) [reply .env.template:11]: GRAFANA_LINK - different per env

File: apps/ai-dial-admin/src/utils/time-filter/get-chart-resolution.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  unit: 'm' | 'h' | 'd';
+}
+
+// Nice bucket sizes in minutes, ascending
+const NICE_BUCKETS_MINUTES = [1, 2, 5, 10, 15, 30, 60, 120, 180, 360, 720, 1440];
+
+const DEFAULT_MAX_DATA_POINTS = 200;
+
+const getMaxDataPoints = (): number => {
+  const env = process.env.NEXT_PUBLIC_CHART_MAX_DATA_POINTS;
```
Comment (PolinaGurinovich97) [inline apps/ai-dial-admin/src/utils/time-filter/get-chart-resolution.ts]: NEXT_PUBLIC_CHART_MAX_DATA_POINTS - please, explain why `NEXT_PUBLIC`

File: apps/ai-dial-admin/src/utils/time-filter/get-chart-resolution.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  if (selectedMinutes >= 1440 && selectedMinutes % 1440 === 0) {
+    return { value: selectedMinutes / 1440, unit: 'd' };
+  }
+  if (selectedMinutes >= 60 && selectedMinutes % 60 === 0) {
+    return { value: selectedMinutes / 60, unit: 'h' };
+  }
+  return { value: selectedMinutes, unit: 'm' };
+};
+
+const WINDOW_PATTERN = /window\(_time,\s*\d+,\s*'[mhd]'\)/g;
```
Comment (PolinaGurinovich97) [inline apps/ai-dial-admin/src/utils/time-filter/get-chart-resolution.ts]: please, explain

**Review by PolinaGurinovich97 — CHANGES_REQUESTED** [review]

File: apps/ai-dial-admin/src/utils/time-filter/get-chart-resolution.ts (hunk 3 above)
Comment (oleksii-donets) [reply apps/ai-dial-admin/src/utils/time-filter/get-chart-resolution.ts]: removed

File: apps/ai-dial-admin/src/utils/time-filter/get-chart-resolution.ts (hunk 2 above)
Comment (oleksii-donets) [reply apps/ai-dial-admin/src/utils/time-filter/get-chart-resolution.ts]: removed

File: .env.template (hunk 1 above)
Comment (oleksii-donets) [reply .env.template:11]: env var NEXT_PUBLIC_CHART_MAX_DATA_POINTS was removed

**Review by dearsimanovich-epam — APPROVED** [review]

**Review by PolinaGurinovich97 — APPROVED** [review]
